packages/gitlab-backend/src/service/router.ts

```diff
@@ -46,7 +46,29 @@ export async function createRouter(
         return req.method === 'GET';
     };
 
-    for (const { host, apiBaseUrl, token } of gitlabIntegrations) {
+    const postFilter = (_pathname: string, req: IncomingMessage): boolean => {
+        if (req.headers['authorization']) delete req.headers['authorization'];
+        return req.method === 'POST';
```

Comment:
Finally, we need to verify that the `graphql` type is `query` and disregard `mutation` and other types to prevent security risks.

Reply:
Fixed and covered by tests